### ISSUE TITLE
Add keyword/phrase search

### ISSUE THREAD

**Issue by KingAkeem** [DedSecInside/TorBot#317]
When crawling links, add the ability to only list links that contain a certain keyword or phrase then display this instead of the default metadata.

**Comment by KingAkeem** [comment]
Updates?

**Comment by apurvabanka** [comment]
@KingAkeem Is there a requirement for this feature? I would like to contribute.

**Comment by PSNAppz** [comment]
@apurvabanka Yes, this is a very interesting/curated feature to have for an OSINT tool. Feel free to open a PR. Also, before that you can discuss the plan here.

**Comment by KingAkeem** [comment]
@apurvabanka No one is currently working on it so feel free as @PSNAppz says

**Comment by KingAkeem** [comment]
@apurvabanka If you need guidance on the code, you can join our Slack channel, and I'll help you.

**Comment by apurvabanka** [comment]
@KingAkeem  Trying to build the docker image using the command 

![image](https://github.com/user-attachments/assets/5090d9ec-0512-498b-9a0e-b803d6f0fab5)

Getting the below error

![image](https://github.com/user-attachments/assets/27524e3b-27f1-4d7f-8510-463597d77052)

Am I missig something?

**Comment by KingAkeem** [comment]
It looks like this merge request is still open https://github.com/DedSecInside/TorBot/issues/346. @PSNAppz made some changes recently that broke the build, I'll have to go through and fix it at some point. Running using another method like `poetry` may work if that wasn't removed also.

**Comment by KingAkeem** [comment]
@apurvabanka The dependency issues have been resolved.

**Comment by DylanM5** [comment]
Has any progress been made on this?
